### DISCUSSION TITLE
What to do when you realize you've been working on the wrong branch

### DISCUSSION THREAD

**Discussion by TaylorHardaway** [nss-evening-cohort-14/evening-client-side#36] [Help Me!] [answered]
Taylor Hardaway

There is one other ticket that talks about undoing a git add. This is actually part of what I should have done, but doesn't answer the whole question.  I actually just left the git add alone and deleted everything and started over on the setup branch. The only weird thing that happened since I did two git commit's was the commit message came back as "setup" instead of what I wrote either time. 

I was just working on one my last repo's for the 30 repos assignment and then realized that I never created a setup branch. My temporary solution was just to delete everything and then redo it on the setup branch. For the purpose of this assignment it wasn't much and the only strange thing that happened was the commit message changing to "setup" since I didn't undo the git add or git commit.

https://stackoverflow.com/questions/2941517/how-to-fix-committing-to-the-wrong-git-branch
https://www.xspdf.com/resolution/50460592.html
https://softwarebrothers.co/blog/5-git-tricks-that-you-can-use-with-git-branches/

I just deleted everything and then created the setup branch and then redid it on the setup branch. I don't actually have any current problems, but I know this wasn't the best way to do this. The above articles seem to address the issue ok, but talk over my head a little. Also, it seems like a common problem so I'm sure it will happen again.

I can't remember which repo it was this morning, but I already merged the branch.

**Comment by drteresavasquez** [comment, marked as answer]
So, the instructions on the assignment say if you run into an issue, just start a new repo and add an additional one to your count. It's so important to read the instructions so you don't get into a rabbit hole.
